### ISSUE TITLE
Fix Request: InconsistentEqualsAndHashCode false positive with field.hashCode

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#105]
This has a false positive:
```
public class BadFromIssue {
    private String field1;
    private String field2;

    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        BadFromIssue that = (BadFromIssue) o;
        return Objects.equals(field1, that.field1)
        && Objects.equals(field2, that.field2);
    }

    public int hashCode() {
        int result = field1.hashCode();
        result = 31 * result + field2.hashCode();
        return result;
    }
}
```
